PLAT-61278: Update Moonstone docs for BodyText through Divider

ryanjduffy opened the pull request:
Updated component docs for consistency.

* Each "base" component includes a short description and a comment directing users toward the "ready to use" component for usage.
* Each "ready to use" component includes a short description, a purpose paragraph, and a usage example.

Enact-DCO-1.0-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

viodragon2 commented on packages/moonstone/DaySelector/DaySelectorDecorator.js:
```diff
@@ -91,9 +90,11 @@ const DaySelectorDecorator = hoc((config, Wrapped) => {
-			 * indexed.
+			 * Called when an day is selected or unselected.
+			 *
+			 * The event payload will be an object with the following members:
```
@webOS101 what's the proper way to document event payloads if we don't want to use `@typedef`?

viodragon2 commented on packages/moonstone/Dialog/Dialog.js:
```diff
@@ -1,15 +1,16 @@
+ * Moonstone styled modal dialog components.
  *
  * @module moonstone/Dialog
+ * @exports	Dialog
```
nit: space

ryanjduffy commented on packages/moonstone/Button/Button.js:
```diff
@@ -52,7 +55,7 @@ const ButtonBase = kind({
 
 		/**
-		 * The color of the underline beneath button's content. Used for `IconButton`.
+		 * The color of the underline beneath button's content.
```
We specifically removed it from the sampler (in #1447) because it can cause some false bug reports.

webOS101 replied:
We could possibly tighten this up to 'Underline color beneath the button's content'.  I think adding  'the' at the front of these descriptions is probably redundant.  I was going to mention that in a few places below.

ryanjduffy commented on packages/moonstone/ContextualPopupDecorator/ContextualPopup.js:
```diff
@@ -51,8 +50,9 @@ const ContextualPopupRoot = Skinnable(
 /**
- * {@link moonstone/ContextualPopupDecorator.ContextualPopup} is a modal component that
- * appears in context to an activator.
+ * A Moonstone styled chrome component used by
```
Something that goes around another thing. :) In the context of React, it'd be the components that wrap around `children`. It's stolen from [Enyo vernacular](https://github.com/enyojs/enyo/blob/master/docs/external.jsdoc#L46) but I should non-Enyo it a bit for clarity.

webOS101 commented on packages/moonstone/Checkbox/Checkbox.js:
```diff
@@ -30,20 +38,23 @@ import componentCss from './Checkbox.less';
-		css: PropTypes.object
+	propTypes: /** @lends moonstone/Checkbox.Checkbox.prototype */ {
+		/**
+		 * The icon displayed when `selected`.
```
Like this one.

webOS101 commented on packages/moonstone/ContextualPopupDecorator/ContextualPopup.js:
```diff
@@ -102,36 +99,38 @@ const ContextualPopupBase = kind({
 
 		/**
-		 * When `true`, the close button is shown; when `false`, it is hidden.
+		 * Shows the close button
```
Maybe 'Adds a close button to the Popup'?  I guess because the word 'show' is in the prop name, there's good reason to use it.

webOS101 commented on packages/moonstone/DayPicker/DayPicker.js:
```diff
@@ -160,13 +140,65 @@ const DayPickerDecorator = compose(
+const DayPicker = DayPickerDecorator(DayPickerBase);
+
+/**
+ * Sets the "aria-label" for the component.
```
This could be simplified to 'The "aria-label" for the component.'

webOS101 commented on packages/moonstone/BodyText/BodyText.js:
```diff
@@ -77,15 +80,21 @@ const BodyTextDecorator = compose(
  *
  * @class BodyText
  * @memberof moonstone/BodyText
+ * @extends moonstone/BodyText.BodyTextBase
```
Bummer.  It seems it was missing from the `moonstone/Button`, which I pulled the example from.  We'll want to update that.

ryanjduffy commented:
I've addressed most of the comments here. A couple general responses:

* In places where I've included standalone prop docs (e.g. `defaultOpen`), I haven't referenced the source HOC because it isn't meaningful. For example, it doesn't matter that `defaultOpen` is provided by `ui/Toggleable` because the prop is configured via HOC config and the docs aren't meaningful.
* The boolean and event prop docs are in line with our doc guidelines (which have admittedly been in a state of flux). We will need to update the rest to match this style.
* Regarding `@typedef`, maybe we hold off on this for now in the interest of moving ahead?
